packages/backend/src/routers/rating.ts

```diff
@@ -27,7 +29,27 @@ export const ratingsRouter = t.router({
                 anonymousIdentifier: await ctx.env.anonymousIdDao.getIdentifier(),
             };
 
-            const analyzedScores = await ctx.env.ratingAnalyzer.analyzeRaring(pendingRating);
+            // Abuse protection: Check if the same rating text has already been submitted for the same professor
+            const professor = await ctx.env.kvDao.getProfessor(input.professor);
+
+            const existingRating = Object.values(professor.reviews)
+                .flat()
+                .find(
+                    (rating) =>
+                        rating.rating === pendingRating.rating &&
+                        rating.anonymousIdentifier === pendingRating.anonymousIdentifier,
```

Comment:
Honestly if someone copies ratings it should be blocked regardless of identifier